File: app/src/main/java/com/nealgosalia/timetable/fragments/ThursdayFragment.java
package com.nealgosalia.timetable.fragments;


import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.nealgosalia.timetable.R;
import com.nealgosalia.timetable.adapters.LecturesAdapter;
import com.nealgosalia.timetable.utils.DividerItemDecoration;
import com.nealgosalia.timetable.utils.FragmentDatabase;
import com.nealgosalia.timetable.utils.Lecture;

import java.util.ArrayList;
import java.util.List;



public class ThursdayFragment extends Fragment {

    public static List<Lecture> lecturesList = new ArrayList<>();
    private RecyclerView recyclerLectures;
    public static LecturesAdapter mLectureAdapter;
    private View v;

    public ThursdayFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        v=inflater.inflate(R.layout.fragment_thursday, container, false);
        FragmentDatabase fragmentDatabase=new FragmentDatabase();
        lecturesList=new ArrayList<>(fragmentDatabase.getLectureList(3,getActivity()));
        recyclerLectures=(RecyclerView)v.findViewById(R.id.listThursday);
        mLectureAdapter = new LecturesAdapter(lecturesList);
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getActivity());
        recyclerLectures.setLayoutManager(mLayoutManager);
        recyclerLectures.setItemAnimator(new DefaultItemAnimator());
        recyclerLectures.addItemDecoration(new DividerItemDecoration(getContext(), LinearLayoutManager.VERTICAL));
        recyclerLectures.setAdapter(mLectureAdapter);
        return v;
    }

}

File: app/src/main/java/com/zeeshanhussain/timetable/ui/fragments/ThursdayFragment.kt
package com.zeeshanhussain.timetable.ui.fragments

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.zeeshanhussain.timetable.R
import com.zeeshanhussain.timetable.adapters.LecturesAdapter
import com.zeeshanhussain.timetable.database.AppDatabase
import com.zeeshanhussain.timetable.model.Lecture
import com.zeeshanhussain.timetable.utils.AppExecutors
import com.zeeshanhussain.timetable.utils.DividerItemDecoration
import com.zeeshanhussain.timetable.utils.RecyclerItemClickListener
import com.zeeshanhussain.timetable.viewmodel.LectureViewModel
import com.zeeshanhussain.timetable.viewmodel.LectureViewModelFactory

import java.util.ArrayList
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class ThursdayFragment : Fragment() {

    private val lecturesList = ArrayList<Lecture>()
    private var mLectureAdapter: LecturesAdapter? = null
    private var placeholderText: TextView? = null
    private var appDatabase: AppDatabase? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_thursday, container, false)
        placeholderText = view.findViewById(R.id.thursdayPlaceholderText)
        appDatabase = AppDatabase.getsInstance(activity)
        val recyclerLectures = view.findViewById<RecyclerView>(R.id.listThursday)
        mLectureAdapter = LecturesAdapter(lecturesList)
        val mLayoutManager = LinearLayoutManager(activity)
        recyclerLectures.layoutManager = mLayoutManager
        recyclerLectures.itemAnimator = DefaultItemAnimator()
        recyclerLectures.addItemDecoration(DividerItemDecoration(context!!, LinearLayoutManager.VERTICAL))
        recyclerLectures.adapter = mLectureAdapter
        val lectureViewModelFactory = LectureViewModelFactory(appDatabase!!, 3)
        val lectureViewModel = ViewModelProviders.of(this, lectureViewModelFactory).get(LectureViewModel::class.java)
        lectureViewModel.lecture.observe(this, Observer { lectures ->
            lecturesList.clear()
            lecturesList.addAll(lectures!!)
            mLectureAdapter!!.notifyDataSetChanged()
            if (lecturesList.size != 0) {
                placeholderText!!.visibility = View.GONE
            }
        })
        recyclerLectures.addOnItemTouchListener(RecyclerItemClickListener(context!!, object : RecyclerItemClickListener.OnItemClickListener {
            override fun onItemClick(view: View, position: Int) {
                val lecture = lecturesList[position]
                showDeleteDialog(lecture, position)
            }
        }))
        return view
    }

    private fun showDeleteDialog(lecture: Lecture, position: Int) {
        val dialogBuilder = AlertDialog.Builder(context)
        dialogBuilder.setTitle(resources.getString(R.string.delete))
        dialogBuilder.setMessage(resources.getString(R.string.delete_lecture))
        dialogBuilder.setPositiveButton(resources.getString(R.string.yes)) { dialog, whichButton ->
            AppExecutors.instance.diskIO().execute {
                appDatabase!!.lectureDao().delete(lecture)
                lecturesList.removeAt(position)
                activity!!.runOnUiThread { mLectureAdapter!!.notifyDataSetChanged() }
            }
        }
        dialogBuilder.setNegativeButton(resources.getString(R.string.no), null)
        val b = dialogBuilder.create()
        b.show()
    }
}